### PR TITLE
Add two new jsp pages to device flow support

### PR DESCRIPTION
**Pull request by ayeshajay**
Purpose
Since device flow needs a page to insert user code and instruct the user at the end of the flow, two new jap pages needed.

Screenshot from 2019-12-03 13-28-29
Screenshot from 2019-12-03 13-28-24
Screenshot from 2019-12-03 13-28-20

https://github.com/wso2/carbon-identity-framework/pull/2508

### CODE REVIEW

File: apps/authentication-portal/src/main/webapp/device-success.jsp [hunk 1]
```diff
@@ -0,0 +1,186 @@
+<!doctype html>
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/device-success.jsp]: Missing license header. Please add it.

File: apps/authentication-portal/src/main/webapp/device-success.jsp [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+<html>
+
+<head>
+    <!-- header -->
+    
+    
+
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/device-success.jsp]: Remove unwanted spaces/new lines and format the code

File: apps/authentication-portal/src/main/webapp/device-success.jsp [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        }
+        
+        body .product-title .product-title-text {
+            margin-top: 0;
+        }
+    </style>
+
+
+
+
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/device-success.jsp]: Use header include instead

File: apps/authentication-portal/src/main/webapp/device-success.jsp [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+</head>
+
+<style>
+    
+    #Para1{
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/device-success.jsp]: Use better id name. And try to use existing styles from the theme or if there is not matching one move this to header include file.

File: apps/authentication-portal/src/main/webapp/enter-user-code.jsp [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+                document.write(new Date().getFullYear());
+            </script>
+            <a href="http://wso2.com/" target="_blank">
+                WSO2
+                Inc
+            </a>
+            . All rights reserved
+        </p>
+    </div>
+</footer>
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/enter-user-code.jsp]: Use product-footer include instead.

File: apps/authentication-portal/src/main/webapp/enter-user-code.jsp [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+<script src="libs/jquery_3.4.1/jquery-3.4.1.js"></script>
+<script src="libs/theme/semantic.min.js"></script>
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/enter-user-code.jsp]: Use footer include instead

File: apps/authentication-portal/src/main/webapp/enter-user-code.jsp [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+                        </path>
+                    </svg>
+                </div>
+                <h1 class="product-title-text">Identity Server</h1>
+            </div>
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/enter-user-code.jsp]: Use product-title include instead

File: apps/authentication-portal/src/main/webapp/enter-user-code.jsp [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+
+    var userCode;
+
+    function loadFunc() {
+
+        const urlParams = new URLSearchParams(window.location.search);
+        userCode = urlParams.get('user_code');
+        document.getElementById("user_code").value = userCode;
+    }
+</script>
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/enter-user-code.jsp]: Move this script end of the body tag

**Comment by ayeshajay** [comment]
Screenshot from 2019-12-03 11-28-42
Screenshot from 2019-12-03 11-28-58

File: apps/authentication-portal/src/main/webapp/device-success.jsp [hunk 9]
```diff
@@ -0,0 +1,103 @@
+<%--
+  ~ Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/device-success.jsp]: Fix the year in license header. It has to be `2019`

File: apps/authentication-portal/src/main/webapp/device-success.jsp [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+        File footerFile = new File(getServletContext().getRealPath("extensions/footer.jsp"));
+        if (footerFile.exists()) {
+    %>
+    <jsp:include page="extensions/footer.jsp"/>
+    <% } else { %>
+    <jsp:directive.include file="includes/footer.jsp"/>
+    <% } %>
+
+    <script type="text/javascript">
+
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/device-success.jsp]: Remove this new line

File: apps/authentication-portal/src/main/webapp/device-success.jsp [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+    <% } else { %>
+    <jsp:directive.include file="includes/footer.jsp"/>
+    <% } %>
+
+    <script type="text/javascript">
+    
+        var userCode;
+    
+        function loadFunc() {
+
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/device-success.jsp]: Remove this new line

File: apps/authentication-portal/src/main/webapp/enter-user-code.jsp [hunk 12]
```diff
@@ -0,0 +1,118 @@
+<%--
+  ~ Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/enter-user-code.jsp]: Fix the year in license header. It has to be `2019`

**Comment by ayeshajay** [comment]
Screenshot from 2019-12-03 12-11-10
Screenshot from 2019-12-03 12-08-03

File: apps/authentication-portal/src/main/webapp/device-success.jsp [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+                    </div>
+                </div>
+            <% } else { %>
+                <div class="ui negative message">
+                    <div class="header">
+                        Error
+                    </div>
+                <p>Requested parameter application name is not present.</p>
+                </div>
+            <% } %>
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/device-success.jsp]: Revert the `ui segment` wrapper which was there.

Comment (ayeshajay) [reply apps/authentication-portal/src/main/webapp/device-success.jsp]: done

**Comment by ayeshajay** [comment]
Screenshot from 2019-12-03 13-28-29
Screenshot from 2019-12-03 13-28-24
Screenshot from 2019-12-03 13-28-20

**Comment by jeradrutnam** [comment]
Please send another PR with localized text. Will merge this PR.

**Pull request merged**